web/search.dart

```diff
@@ -203,24 +211,56 @@ void initializeSearch(
         query, "<strong class='tt-highlight'>$sanitizedText</strong>");
   }
 
+  var categoriesMap = <String, Element>{};
+
+  // Maps a suggestions library/class to the other suggestions that search found if any.
+  void categorize(Element suggestionLibrary, Element suggestion) {
+    var input = suggestionLibrary.innerHtml;
+
+    if (input == null) {
+      return;
+    }
+
+    var element = categoriesMap[input];
+    if (element != null) {
+      element.append(suggestion);
+      categoriesMap.update(input, (value) => element);
```

Comment:
Since you're not using `value` here, I think it's easiest to just write `categoriesMap[input] = element;` like you do on line 230.